frame/executive/src/lib.rs

```diff
@@ -496,53 +548,71 @@ where
 		sp_io::init_tracing();
 		sp_tracing::within_span! {
 			sp_tracing::info_span!("execute_block", ?block);
+			// Execute `on_runtime_upgrade` and `on_initialize`.
+			let mode = Self::initialize_block(block.header());
+			let num_inherents = Self::initial_checks(&block) as usize;
+			let (header, extrinsics) = block.deconstruct();
 
-			Self::initialize_block(block.header());
+			if mode == ExtrinsicInclusionMode::OnlyInherents && extrinsics.len() > num_inherents {
```

Comment:
I think the naming around this is a best effort, but still somewhat broken. Something that's called "Extrinsic" inclusion mode will soon dictate whether or not `on_idle` will execute or not. Broken :D

Reply:
Yea i will probably revert to `RuntimExecutionMode`... (there was a discussion above about it).